CMakeLists.txt

```diff
@@ -1,69 +1,145 @@
-    # Note from hipsycl: "Do not call target_sources after add_sycl_to_target or
-    # dependency tracking on compiler flags will break in subtle ways"
-    function(add_sycl_to_target)
+#Makes it easy to install + adds the files to the INCLUDE property of the lib
+#i.e. don't need target_include_dir.. also no GLOBS
+target_sources(NESO-Particles
+    PUBLIC
+    FILE_SET public_headers
+    TYPE HEADERS
+    BASE_DIRS ${CMAKE_CURRENT_SOURCE_DIR}/include
+    FILES
+    ${CMAKE_CURRENT_SOURCE_DIR}/include/cell_dat.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/include/access.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/include/boundary_conditions.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/include/cartesian_mesh.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/include/cell_binning.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/include/cell_dat_compression.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/include/cell_dat.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/include/cell_dat_move.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/include/cell_dat_move_impl.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/include/communication.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/include/compute_target.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/include/containers
+    ${CMAKE_CURRENT_SOURCE_DIR}/include/departing_particle_identification.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/include/departing_particle_identification_impl.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/include/domain.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/include/global_mapping.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/include/global_mapping_impl.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/include/global_move_exchange.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/include/global_move.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/include/local_mapping.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/include/local_move.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/include/loop
+    ${CMAKE_CURRENT_SOURCE_DIR}/include/mesh_hierarchy.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/include/mesh_interface.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/include/mesh_interface_local_decomp.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/include/neso_particles.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/include/packing_unpacking.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/include/parallel_initialisation.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/include/particle_dat.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/include/particle_group.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/include/particle_io.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/include/particle_remover.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/include/particle_set.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/include/particle_spec.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/include/particle_sub_group.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/include/profiling.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/include/typedefs.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/include/utility.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/include/utility_mesh_hierarchy_plotting.hpp
+    )
 
-    endfunction()
-  endif()
 
-  if(NESO_PARTICLES_DEVICE_TYPE STREQUAL GPU)
+#Don't like this .... TODO: FIXME:
+#Should be a runtime thing?
```

Comment:
These define how the parallel loops happen for the old-style particle particle loops - ideally these are deprecated and phased out completely.